cron/format/testdata/check3.json

```diff
@@ -0,0 +1,51 @@
+{
+   "date": "2021-08-25",
+   "repo": {
+      "name": "org/name",
+      "commit": "68bc59901773ab4c051dfcea0cc4201a1567ab32"
+   },
+   "scorecard": {
+      "version": "1.2.3",
+      "commit": "ccbc59901773ab4c051dfcea0cc4201a1567abdd"
+   },
+   "checks": [
+      {
+         "details": [
+            "Warn: warn message: bin/binary.elf"
+         ],
+         "score": 0,
+         "reason": "min result reason",
+         "name": "Check-Name",
+         "documentation": {
```

Comment:
These will be very repetitive in the BQ table. Is there any benefit to including this in every single row, as opposed to asking users to get this info from elsewhere?

Reply:
yep, I agree and thought about this too. BQ table supports column compression (Run Length Encoding, etc), so this should not take much space on disk. However, I just looked up pricing and we're charged based on [data size before compression](https://cloud.google.com/bigquery/pricing#data), so thats not ideal. 
The main reason I included it is ease for users to retrieve and for us to keep scorecard changes in sync with where the URL lives. Deps.dev already broke because the doc location changed.

Alternatives:
1. Create a url per json instead of per check. That still requires us to keep url#check-name non-breaking. This would break if we separate checks in their own doc file (as suggested in https://github.com/ossf/scorecard/issues/998).
2. URL that redirects to doc...

What are better options?

Reply:
@azeemsgoogle any thoughts on this?

Reply:
Do we want to launch Scorecard V2 (version with scores)? Given our conversations with ProjectsByIf, it is likely that our result formats will undergo massive overhaul soon. Do we gain anything by launching this version then?

Reply:
It helps users of deps.dev get a better format. I don't know how much traffic they get, but I think they are a source of discovery for scorecard we should not neglect. @kimsterv do you know?

In terms of projectsbyif, if we need to change scorecard/format a lot, then this means it will take quite some time to land the changes. In the meantime it's useful to have deps.dev use a score-based version.

wdut?

Reply:
follow-up about pricing. https://cloud.google.com/bigquery/pricing says, under storage section: `$0.020 per GB`. If we reach 1M repos, considering a `300` characters/bytes per doc (url + description), given 10 checks, and with 4 runs per month, we get:
$0.020 * 300 * 10 * 1000000 * 4 / 1000000000 = $0.24/month.

Please chime in if I forgot something in the calculation above. In the calculation is correct, pricing is not an issue/blocker.

Given the use of compression by BQ, latency should not be an issue either.

So I think we can keep the doc fields per check in JSON. 

@oliverchang waiting for ur feedback

Reply:
My concern is making deps.dev and OSSF metrics teams go through 2 different migrations. There might also be other teams (like Project Thoth - https://github.com/ossf/scorecard/issues/978#issuecomment-917955571) who are relying on the BQ data without us knowing about it. It might be worth figuring out a plan which minimizes the overhead on our clients for doing these migrations.  

re: pricing/storage - IMO its fine to have this data, even though it maybe redundant. For teams relying on our BQ data, it'll be useful to have this present in BQ itself.

Reply:
Thanks for explaining. Ease of use + link stability seems like a good enough reason to keep this in the table.

Reply:
re: other teams relying on BQ dataset. I think we need to announce breaking changes/new versions on a dedicated channel. Maybe scorecard-announce@ mailing list or a dedicated section on pkg.go.dev, probably both. We cannot, ourselves, keep an inventory of *who* is using scorecard. I've added an item for the next scorecard sync. 

Note that updating the cron JSON does not force our users to use it. We can still support the old format for 6 months.